context_propagation/say_hello_workflow.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require 'temporalio/workflow'
+require_relative 'say_hello_activity'
+
+module ContextPropagation
+  class SayHelloWorkflow < Temporalio::Workflow::Definition
+    def execute(name)
+      Temporalio::Workflow.logger.info("Workflow called by user: #{Thread.current[:my_user]}")
+
+      # Wait for signal then run activity
+      Temporalio::Workflow.wait_condition { @complete }
```

Comment:
Neat. But, won't the `logger.info` statement be called every time the condition is evaluated? That would be particularly undesirable since it is saying that a user query is being handled.

Reply:
Oh, no, maybe I don't understand what `@complete` is. I thought it was a reference to the method which is used as a query hander. What is it?

Reply:
> I don't understand what @complete is

`@complete` is a Ruby attribute. That is class state, akin to `self.complete` in Python. However, unlike Python, accessing an uninitialized Ruby attribute returns `nil` instead of errors. Also unlike Python, Ruby attributes are only visible within the class (i.e. they are "private") and you have to use other things to expose them.

> Does `@foo` just always evaluate to `nil` for any value of "foo" unless it's been set?

Yup, which makes Ruby workflows so nice and easy to write and really Ruby is a very pretty language altogether, so much so that in my biased opinions, workflows written in Ruby will be the most dev friendly to read/write of all.

Reply:
I agree they're very nice to read. I don't think that having a poor typing story is the most dev-friendly for writing.